src/components/wp-version-selector/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { SelectControl, Icon } from '@wordpress/components';
+import { warning } from '@wordpress/icons';
+import { useI18n } from '@wordpress/react-i18n';
+import offlineIcon from 'src/components/offline-icon';
+import { Tooltip } from 'src/components/tooltip';
+import { useOffline } from 'src/hooks/use-offline';
+import { cx } from 'src/lib/cx';
+import { isWordPressDevVersion } from 'src/lib/version-utils';
+import { isWordPressBetaVersion } from 'src/lib/wordpress-version-utils';
+import { useGetWordPressVersions } from 'src/stores/wordpress-versions-api';
+import { DEFAULT_WORDPRESS_VERSION } from 'vendor/wp-now/src/constants';
+import { addWpVersionToList } from './add-wp-version-to-list';
+
+type WPVersionSelectorProps = {
+	selectedValue: string;
+	onChange: ( version: string ) => void;
+	errorMessage?: string | null;
+	disabled?: boolean;
+	/** Is used if you want to add a custom option to the list, for example the current version of a site */
+	extraOptions?: { label: string; value: string }[];
+	/** Fallback options shown when available versions couldn't be fetched */
+	fallbackOptions: { label: string; value: string }[];
+};
+
+export const WPVersionSelector = ( {
+	selectedValue,
+	onChange,
+	errorMessage,
+	disabled = false,
+	extraOptions,
+	fallbackOptions,
+}: WPVersionSelectorProps ) => {
+	const { __ } = useI18n();
+	const isOffline = useOffline();
+	const offlineMessage = __( 'Changing WordPress version requires an internet connection.' );
+	const { data: wpVersions = [] } = useGetWordPressVersions();
+
+	let betaVersions: { label: string; value: string }[] = wpVersions.filter(
+		( version ) => version.isBeta || version.isDevelopment
+	);
+	let stableVersions: { label: string; value: string }[] = wpVersions.filter(
+		( version ) => ! version.isBeta && ! version.isDevelopment && version.value !== 'latest'
+	);
+	extraOptions?.forEach( ( extraOption ) => {
+		const alreadyExists = wpVersions.some( ( version ) => version.value === extraOption.value );
+		if ( alreadyExists ) {
+			return;
+		}
+
+		if (
+			isWordPressBetaVersion( extraOption.value ) ||
+			isWordPressDevVersion( extraOption.value )
+		) {
+			betaVersions = addWpVersionToList( extraOption, betaVersions );
+		} else {
+			stableVersions = addWpVersionToList( extraOption, stableVersions );
+		}
+	} );
+
+	return (
+		<label className="flex flex-1 flex-col gap-1.5 leading-4">
+			<span className="font-semibold flex items-center gap-2">
+				{ __( 'WordPress version' ) }
+				{ selectedValue !== 'latest' && (
+					<Tooltip
+						text={ __( 'WordPress Core automatic updates will be disabled for this site.' ) }
+						placement="top"
+					>
+						<Icon icon={ warning } className="text-[#ae5c00]" size={ 16 } />
```

Comment:
I don't think this color is required. I don't see any change in the UI after removing it.

```suggestion
						<Icon icon={ warning } size={ 16 } />
```

![Screenshot 2025-04-21 at 11 52 02](https://github.com/user-attachments/assets/dc0b5b9a-e6d6-4e92-96e7-efc4ed9dd90a)